File: module-user/src/main/java/club/ccit/user/MyAdapter.java
package club.ccit.user;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import club.ccit.basic.BaseAdapter;
import club.ccit.common.LogUtils;
import club.ccit.sdk.demo.NewsListBean;
import club.ccit.user.databinding.ItemMyBinding;


/**
 * FileName: TestAdapter
 *
 * @author: 瞌睡的牙签
 * Date: 2021/12/3 1:17 下午
 * Description:
 * Version:
 */
public class MyAdapter extends BaseAdapter<ItemMyBinding> {

    /**
     * 实例化
     * @param list
     */
    public MyAdapter(List list) {
        this.list = list;
    }

    /**
     * 绑定数据
     * @param position
     */
    @Override
    protected void onBindingViewData(int position) {
        NewsListBean.Result bean = (NewsListBean.Result) list.get(position);
        // 设置文本数据
        binding.titleTextView.setText(position+1+"  "+bean.getArticle_title());
        binding.timeTextView.setText(bean.getArticle_date());
        binding.contentTextView.setText(bean.getArticle_text());
        binding.authorTextView.setText(bean.getArticle_user()+"/文");
        binding.itemCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LogUtils.i(position+1+"  "+bean.getArticle_title());
            }
        });
    }

    /**
     * 绑定视图
     * @param parent
     * @return
     */
    @Override
    protected ItemMyBinding getViewBinding(ViewGroup parent) {
        return ItemMyBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
    }

}

File: module-home/src/main/java/club/ccit/home/fragment/MyAdapter.kt
package club.ccit.home.fragment

import club.ccit.basic.KtBaseAdapter
import club.ccit.common.LogUtils
import android.view.ViewGroup
import android.view.LayoutInflater
import club.ccit.home.databinding.ItemMyBinding
import club.ccit.sdk.demo.NewsListBean

/**
 * FileName: TestAdapter
 *
 * @author: 瞌睡的牙签
 * Date: 2021/12/3 1:17 下午
 * Description:
 * Version:
 */
class MyAdapter(list: MutableList<*>?) : KtBaseAdapter<ItemMyBinding?>() {
    /**
     * 绑定数据
     * @param position
     */
    override fun onBindingViewData(position: Int) {
        val bean = list!![position] as NewsListBean.Result
        // 设置文本数据
        binding!!.titleTextView.setText(bean.article_title)
        binding!!.timeTextView.text = bean.article_date
        binding!!.contentTextView.text = bean.article_text
        binding!!.authorTextView.text = bean.article_user + "/文"
        binding!!.itemCardView.setOnClickListener { LogUtils.i("  " + bean.article_title) }
    }

    /**
     * 绑定视图
     * @param parent
     * @return
     */
    override fun getViewBinding(parent: ViewGroup?): ItemMyBinding {
        return ItemMyBinding.inflate(LayoutInflater.from(parent!!.context), parent, false)
    }

    /**
     * 实例化
     * @param list
     */
    init {
        this.list = list
    }
}